src/Microsoft.OpenApi.Readers/OpenApiTextReaderReader.cs

```diff
@@ -50,7 +51,12 @@ public OpenApiDocument Read(TextReader input, out OpenApiDiagnostic diagnostic)
                 return new OpenApiDocument();
             }
 
-            return new OpenApiYamlDocumentReader(this._settings).Read(yamlDocument, out diagnostic);
+            var asJsonNode = yamlDocument.ToJsonNode();
+
+            diagnostic = null; // TBD
+            return asJsonNode.Deserialize<OpenApiDocument>();
```

Comment:
I honestly don't see how using Deserialize<T> is going to work for us.  Using an automatic deserialization mechanism strongly couples the serialized format to the in-memory model.  We specifically wrote the deserializer classes so we can deserialize any version of OpenAPI into the "latest" in-memory model.  

It is not clear to me from this PR how you are proposing we continue allowing our users to import OpenAPI without caring what version of OAS the input format is.  And to clarify, this is one of our most important features.

Reply:
Strictly speaking, this change isn't necessary because this code will be deleted.

> Using an automatic deserialization mechanism strongly couples the serialized format to the in-memory model.

How so?  I'm doing the actual parsing with YamlSharp; both YAML and JSON are supported through this.  It parses into the `YamlDocument` model, which I then translate to `JsonNode`.  Then I just use the built-in `JsonSerializer` to deserialize it into `OpenApiDocument`.

Going the other way, if you want JSON, you just use `JsonSerializer.Serialize(openApiDoc)`.  If you want YAML, you use [this method](https://github.com/microsoft/OpenAPI.NET/pull/1191/files#diff-83c749e84579435c1783f1791fc5271c6ba649262675829f0c78b2d322d5d335R219) which we'll need to expose as an extension.

> We specifically wrote the deserializer classes so we can deserialize any version of OpenAPI into the "latest" in-memory model.

This supports the same thing.  I can drop a 2.0 or a 3.1 schema into my `DevTest` test and it still deserializes just fine.  Users don't need to care which OpenAPI version they have.

Reply:
How does the JsonDeserializer.Deserialize convert body parameters into a requestBody object?  How does the JsonSerializer.Deserialize know how to convert produces/consumes into media type objects?  We have many of these kinds of translations that System.Text.Json knows nothing about.

Reply:
I don't follow.  The paths use `OpenApiParameter` which isn't associated with `OpenApiRequestBody` anywhere that I can see.

Reply:
In our V2 Deserializer, we convert parameters that are of type `body` or `form` into a request body.  See code here:

https://github.com/microsoft/OpenAPI.NET/blob/vnext/src/Microsoft.OpenApi.Readers/V2/OpenApiOperationDeserializer.cs#L126-L138